dm/syncer/syncer.go

```diff
@@ -858,7 +858,13 @@ func (s *Syncer) updateReplicationLagMetric() {
 	}
 	if minTS != int64(0) {
 		lag = s.calcReplicationLag(minTS)
+		// this means the syncer is blocked so no job is being processed and the lag is always equal to the last one.
+		// we just increased the lag time by 1 to make this metric looks a bit more normal.
+		if lag == s.secondsBehindMaster.Load() {
+			lag += 1
```

Comment:
When syncer get blocked, can dm-worker enter this if block?

Reply:
yes , this func is called by ticker https://github.com/pingcap/tiflow/blob/801bb607615e42be6be7770a302d692bca037834/dm/syncer/syncer.go#L1598

Reply:
What if dml buckets are all empty and syncer gets blocked at a ddl? Will minTS always be `0` and never enter this part?

Reply:
yes, but if in this case, it's hard  to calc lag becasue the first ddl job is blocked, i can't known this job TS

one way to handle this is to add a var in syncer struct used to record the first received job's TS,
but I'm worried that it's not worth losing some performance just to deal with this rare situation, which only happens when there is no DML, only a DDL and the first DDL is Blocked

Reply:
I think just handle for the second and later on DDLs are enough. But I'm afraid after https://github.com/pingcap/tiflow/pull/4491 we can't handle this well

Reply:
cnsidering that this is a relatively rare case (no DML only one pending DDL) maybe we can ignore this case ?
and the lag will be updated after this pengding ddl executed to downstream 😂

Reply:
If we use pessimistic ddl lock, will this happen frequently?

Reply:
i think it's still rare, unless user only use dm to sync DDL, btw can you provide a better way to solve the problem? i would love to imporve the current situation

Reply:
Maybe we should clear these buckets through the heartbeat mechanism: only clear `workerJobTSArray` only when all jobs in this bucket are cleared and handle a heartbeat job.
But after a second thought, it's a bit complicated. The current implementation is also okay.